frontend/src/stimulus/controllers/beforeunload.controller.ts

```diff
@@ -0,0 +1,75 @@
+import { ApplicationController } from 'stimulus-use';
+import { TurboBeforeVisitEvent } from '@hotwired/turbo';
+
+export function setPageWasEdited() {
+  const application = window.Stimulus;
+  const controller = application.getControllerForElementAndIdentifier(document.body, 'beforeunload') as BeforeunloadController;
+  if (controller) {
+    controller.pageWasEdited = true;
+  }
+}
+
+export function setPageWasSubmitted() {
+  const application = window.Stimulus;
+  const controller = application.getControllerForElementAndIdentifier(document.body, 'beforeunload') as BeforeunloadController;
+  if (controller) {
+    controller.pageWasEdited = true;
+  }
+}
+
+export class BeforeunloadController extends ApplicationController {
```

Comment:
Should we use a default export here to be consistent with other Stimulus controllers?

Reply:
This will not work if you want to reference it by type within the same class as with the helpers. If you would only (dynamically) import it, that's fine.